src/main/java/io/cdap/plugin/snowflake/actions/loadunload/unload/UnloadAction.java

```
@@ -0,0 +1,59 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.snowflake.actions.loadunload.unload;
+
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.cdap.etl.api.PipelineConfigurer;
+import io.cdap.cdap.etl.api.StageConfigurer;
+import io.cdap.cdap.etl.api.action.Action;
+import io.cdap.cdap.etl.api.action.ActionContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Unloads data from a Snowflake table (or query) into one or more files.
+ */
+@Plugin(type = Action.PLUGIN_TYPE)
+@Name("SnowflakeToCloudStorage")
+@Description("Unload data from a Snowflake table (or query) into one or more files.")
+public class UnloadAction extends Action {
+  private static final Logger LOG = LoggerFactory.getLogger(UnloadAction.class);
+
+  private final UnloadActionConfig config;
+
+  public UnloadAction(UnloadActionConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void run(ActionContext context) throws Exception {
+    FailureCollector collector = context.getFailureCollector();
+    config.validate(collector);
```

please throw the exception after this